src/dissectors/ec_o5logon.c

```diff
@@ -46,6 +65,24 @@ void o5logon_init(void);
 
 /************************************************/
 
+#undef memrchr
+#define memrchr my_memrchr
```

Comment:
So we'd should only use it when being on a BSD or DARWIN a.k.a. MacOS operating system. The pre-processor variables are set by cmake.
I'd even recommend putting it into a "common" file like ec_utils.c that make it easier to be reused by other functions in future.
Or we could put it into a dedicated source / header file and include it into compilation processes depending on Cmake tests.

I could also help you with the Cmake part if you want.